mr/src/test/java/org/apache/iceberg/mr/TestIcebergInputFormat.java

```diff
@@ -0,0 +1,133 @@
+import org.apache.iceberg.hadoop.HadoopTables;
+import org.apache.iceberg.io.FileAppender;
+import org.apache.iceberg.types.Types;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import static org.apache.iceberg.types.Types.NestedField.optional;
+import static org.apache.iceberg.types.Types.NestedField.required;
+
+
+public class TestIcebergInputFormat {
+  private static final Configuration CONF = new Configuration();
+  private static final HadoopTables TABLES = new HadoopTables(CONF);
+
+  private File tableLocation;
+
+  private static final Schema SCHEMA = new Schema(
+      required(1, "id", Types.LongType.get()),
+      optional(2, "data", Types.StringType.get()),
+      required(3, "date", Types.StringType.get()));
+
+  private static final PartitionSpec PARTITION_BY_DATE = PartitionSpec
+      .builderFor(SCHEMA)
+      .identity("date")
+      .build();
+
+  @Rule
+  public TemporaryFolder temp = new TemporaryFolder();
+  IcebergInputFormat<GenericData.Record> icebergInputFormat;
+
+  @Test
+  public void test() throws IOException, InterruptedException {
+    tableLocation = new File(temp.newFolder(), "table");
+    Table table = TABLES.create(SCHEMA, PARTITION_BY_DATE, tableLocation.toString());
+    List<GenericData.Record> records = RandomAvroData.generate(SCHEMA, 5, 0L);
+    File file = temp.newFile();
+    Assert.assertTrue(file.delete());
+    try (FileAppender<GenericData.Record> appender = Avro.write(Files.localOutput(file))
+                                                         .schema(SCHEMA)
+                                                         .named("avro")
+                                                         .build()) {
+      appender.addAll(records);
+    }
+
+    DataFile dataFile = DataFiles.builder(PARTITION_BY_DATE)
+                                 .withPartition(partitionData("2020-03-15"))
+                                 .withRecordCount(records.size())
+                                 .withFileSizeInBytes(file.length())
+                                 .withPath(file.toString())
+                                 .withFormat("avro")
+                                 .build();
+
+    table.newAppend().appendFile(dataFile).commit();
+
+    Job job = Job.getInstance(new Configuration());
+    IcebergInputFormat
+        .configure(job)
+        .readFrom(tableLocation.getAbsolutePath());
+
+    TaskAttemptContext context = new TaskAttemptContextImpl(new JobConf(job.getConfiguration()), new TaskAttemptID());
+    icebergInputFormat = new IcebergInputFormat<>();
+    List<InputSplit> splits = icebergInputFormat.getSplits(context);
+    final RecordReader<Void, GenericData.Record> recordReader =
+        icebergInputFormat.createRecordReader(splits.get(0), context);
+    recordReader.initialize(splits.get(0), context);
+    while (recordReader.nextKeyValue()) {
+      System.out.println(recordReader.getCurrentValue());
```

Comment:
We should make this an assertion before committing this PR.

Reply:
Tests are not done well. I plan to write better tests before publishing this PR

Reply:
Fixed.